### PR TITLE
GoogleCloudMonitoring: Refactor metricType input

### PR DESCRIPTION
**Pull request by andresmgot**
**What is this feature?**

Remove the `MetricType` input from the API spec since it can be represented as a filter.

**Which issue(s) does this PR fix?**:

Ref #58333

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/tsdb/cloudmonitoring/cloudmonitoring.go [hunk 1]
```diff
@@ -206,6 +206,22 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
 	}
 }
 
+func migrateMetricTypeFilter(metricTypeFilter string, prevFilters interface{}) []string {
+	metricTypeFilterArray := []string{"metric.type", "=", metricTypeFilter}
+	filters := []string{}
```
Comment (asimpson) [inline pkg/tsdb/cloudmonitoring/cloudmonitoring.go:211]: Nit: can we name the return variable `filters` , e.g. `(filters []string)` and then we don't need this here?

File: pkg/tsdb/cloudmonitoring/cloudmonitoring.go [hunk 2]
```diff
@@ -206,6 +206,22 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
...
+	if prevFilters != nil {
+		filtersIface := prevFilters.([]interface{})
+		for _, f := range filtersIface {
+			filters = append(filters, f.(string))
+		}
+		metricTypeFilterArray = append([]string{"AND"}, metricTypeFilterArray...)
+		filters = append(filters, metricTypeFilterArray...)
```
Comment (asimpson) [inline pkg/tsdb/cloudmonitoring/cloudmonitoring.go:218]: Nit: We can  `return`  here directly without assigning to `filters` first?

File: pkg/tsdb/cloudmonitoring/cloudmonitoring.go [hunk 3]
```diff
@@ -206,6 +206,22 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
...
+	} else {
+		filters = metricTypeFilterArray
```
Comment (asimpson) [inline pkg/tsdb/cloudmonitoring/cloudmonitoring.go:220]: Nit: we can `return metricTypeFilterArray` directly without assigning first?

File: pkg/tsdb/cloudmonitoring/cloudmonitoring.go [hunk 4]
```diff
@@ -206,6 +206,22 @@ func newInstanceSettings(httpClientProvider httpclient.Provider) datasource.Inst
...
+	}
+	return filters
```
Comment (asimpson) [inline pkg/tsdb/cloudmonitoring/cloudmonitoring.go:222]: Nit: if we return early in the both branches of the `if` we don't need a `return` here.

**Review by asimpson — APPROVED** [review]
Nice work on this :rocket: 

:heavy_check_mark: tests pass for me locally.

I think we could shorten the mgirate function a bit but that's a nit and I leave it up to you :sunglasses:

File: pkg/tsdb/cloudmonitoring/cloudmonitoring.go (hunk 1 above)
Comment (andresmgot) [reply pkg/tsdb/cloudmonitoring/cloudmonitoring.go:211]: generally I don't like named return variables very much. I found them more difficult to read in bigger functions and prone to bugs (e.g. returning partial results with errors) :thinking:

Comment (asimpson) [reply pkg/tsdb/cloudmonitoring/cloudmonitoring.go]: Works for me :smile: I tend to use them in cases like this where I'm creating a variable to return it later. Like I said though, it's a nit.

**Pull request merged**